external/src/main/java/com/redhat/cloud/policies/engine/LightweightEngineImpl.java

```diff
@@ -0,0 +1,265 @@
+            fullTrigger.getDampenings().add(dampening);
+        } else if (fullTrigger.getDampenings().size() > 1) {
+            // This should never be logged.
+            LOGGER.warnf("Trigger with more than one dampening found: %s", fullTrigger);
+        }
+    }
+
+    /*
+     * Returns alerts created from the fired triggers.
+     * This is kind of the Java equivalent of the old AlertOnSatisfiedDampening rule.
+     * See the ConditionMatch.drl file for more details about that rule.
+     */
+    private Set<Alert> fireTriggers(Event event) {
+        Map<String, FullTrigger> tenantTriggers = triggersByTenant.get(event.getTenantId());
+        if (tenantTriggers == null) {
+            LOGGER.debugf("No triggers found for tenant %s", event.getTenantId());
+            return Collections.emptySet();
+        } else {
+            LOGGER.debugf("Found %d triggers for tenant %s", tenantTriggers.size(), event.getTenantId());
+            Set<Alert> alerts = new HashSet<>();
+            for (FullTrigger fullTrigger : tenantTriggers.values()) {
+                LOGGER.debugf("Starting the trigger conditions evaluation");
+                Set<ConditionEval> conditionEvals = new HashSet<>();
+                for (Condition condition : fullTrigger.getConditions()) {
+                    switch (condition.getType()) {
+                        case EVENT:
+                            evaluateCondition(fullTrigger.getTrigger(), (EventCondition) condition, event).ifPresent(conditionEvals::add);
+                            break;
+                        default:
+                            LOGGER.warnf("Unsupported condition type %s", condition);
+                            break;
+                    }
+                }
+                // The trigger should always have exactly one dampening.
+                Dampening dampening = fullTrigger.getDampenings().get(0);
+                performDampening(dampening, fullTrigger.getTrigger(), conditionEvals);
+                if (dampening.isSatisfied()) {
+                    LOGGER.debugf("Dampening satisfied");
+                    Alert alert = new Alert(event.getTenantId(), fullTrigger.getTrigger(), dampening, dampening.getSatisfyingEvals());
+                    policiesHistoryService.put(alert);
+                    alerts.add(alert);
+                } else {
+                    LOGGER.debugf("Dampening not satisfied");
+                }
+                dampening.reset();
+            }
+            return alerts;
+        }
+    }
+
+    /*
+     * Java equivalent of the old Event rule.
+     * See the ConditionMatch.drl file for more details about that rule.
+     */
+    private Optional<EventConditionEval> evaluateCondition(Trigger trigger, EventCondition eventCondition, Event event) {
+        if (trigger.getMode() == eventCondition.getTriggerMode() && trigger.getSource().equals(event.getDataSource())
+                && eventCondition.getDataId().equals(event.getDataId())) {
```

Comment:
What does it mean that the `EventCondition` and `Event` have different `mode`, `source` or `dataId` ? Is this something we should log?

Reply:
This Java code is a strict translation of the Drools rule. That old rule didn't log anything when these conditions were not met so I don't think we need to log anything.

Reply:
Sounds good!